src/gui/spacemigration.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) by Hannah von Reth <hannah.vonreth@owncloud.com>
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+#include "spacemigration.h"
+
+#include "common/version.h"
+#include "gui/accountmanager.h"
+#include "gui/folderman.h"
+#include "libsync/account.h"
+#include "libsync/graphapi/drives.h"
+
+#include <QJsonArray>
+
+Q_LOGGING_CATEGORY(lcMigration, "gui.migration.spaces", QtInfoMsg)
+
+using namespace OCC;
+
+SpaceMigration::SpaceMigration(const AccountStatePtr &account, const QString &path, QObject *parent)
+    : QObject(parent)
+    , _accountState(account)
+    , _path(path)
+{
+}
+
+void SpaceMigration::start()
+{
+    FolderMan::instance()->setSyncEnabled(false);
+
+    QJsonArray folders;
+    for (auto *f : FolderMan::instance()->folders()) {
+        // same account and uses the legacy dav url
+        // already migrated folders are ignored
+        if (f->accountState()->account() == _accountState->account() && f->webDavUrl() == _accountState->account()->davUrl()) {
+            folders.append(f->remotePath());
+            _migrationFolders.append(f);
+        }
+    }
+    const QJsonObject obj {
+        { QStringLiteral("version"), Version::version().toString() },
+        { QStringLiteral("user"), _accountState->account()->davUser() },
+        { QStringLiteral("remotefolder"), folders }
+    };
+    auto job = new JsonJob(_accountState->account(), _accountState->account()->url(), _path, QByteArrayLiteral("POST"), obj);
+    job->setParent(this);
+    connect(job, &JsonJob::finishedSignal, this, [job, this] {
+        const int status = job->reply()->attribute(QNetworkRequest::HttpStatusCodeAttribute).toInt();
+        if (status == 200 && job->parseError().error == QJsonParseError::NoError) {
+            migrate(job->data().value(QStringLiteral("folders")).toObject());
+        } else {
+            qCInfo(lcMigration) << status << job->data();
```

Comment:
Should not be `info` I guess. A little explanation what this message is about would help, too.

Reply:
Not much to log here, and the logging category should provide enough context.